proposals/build-provenance-and-code-signing-for-homebrew.md

```diff
@@ -110,3 +110,55 @@ For formulae with verifiable source artifacts, the signed bottle attestation des
+
+## Threat model
+
+### Scope
+
+The scope of this proposal is *authenticated build provenance* for Homebrew bottles.
+
+As an end state: when a user performs `brew install foo`, the bottle corresponding to `foo` will be installed *if and only if* it has a valid signature defined over an *attestation* for that bottle. That signature in turn is considered valid *if and only if* it is both cryptographically valid  and* corresponds to the tap repository identity that `foo` was built in (e.g. `homebrew/homebrew-core`).
+
+### Attacker models
+
+All attacker models listed below are remote models. Local attackers are not modeled, under Homebrew's pre-existing operating assumption that a local attacker capable of modifying files or running code has already won. This assumption is comparable to the client integrity assumption in the Web PKI (i.e., that the client's root of trust is not itself compromised).
+
+These models are not intended to be exhaustive.
+
+#### Compromise of bottle storage
+
+**Scenario**: Mallory is able to circumvent Homebrew's ordinary bottle building and publishing workflows and is able to upload malicious bottles to a tap's bottle storage.
+
+**Mitigation**: With *just* upload access to the bottle storage, Mallory is unable to forge valid signatures for her malicious bottles. `brew install $compromised-bottle` fails to install due to a missing or mismatched signature, and users who attempt to install `$compromised-bottle` are loudly notified.
+
+**Outcome**: Mallory is **unable** to force an install of a compromised bottle. Mallory remains undetected.
+
+#### Compromise of bottle-building workflows
+
+**Scenario**: Mallory is able to interpose or otherwise control Homebrew's authentic bottle-building workflows. She is able to modify or outright replace authentic bottles with malicious ones in a way that does not reveal her presence.
+
+**Mitigation**: With *just* access to the bottle-building workflows, Mallory still lacks access to the signing workflow. As such, she is unable to force valid signatures for her malicious bottles and remains unable to compel malicious bottle installation as described above.
+
+**Outcome**: Mallory is **unable** to force an install of a compromised bottle. Mallory remains undetected.
+
+#### Compromise of bottle-signing workflows
+
+**Scenario**: Mallory is able to interpose or otherwise control Homebrew's authentic bottle-signing workflows. She is able to sign for malicious bottles that she controls, resulting in signatures that match a given tap's signing identity but are inauthentic.
+
+**Mitigation**: With *just* access to the bottle-signing workflows, Mallory is unable to upload her inauthentically signed bottles to the bottle storage. Her bottles *would* pass signature verification but remain inaccessible to users. By virtue of Sigstore's transparency services, Mallory loses her stealth in mounting this attack.
```

Comment:
Should we mention in this and the next compromise that as part of the mitigation/detection, Homebrew MUST monitor for instances of core tap identities in the log and alert the ecosystem if unexpected examples are found?

Reply:
Done: I've put this in the summary with references to the relevant attacker models. Let me know if you think the language isn't strong enough; I couched it slightly since monitoring work isn't included in the current scope (although I don't think it'd be difficult to add) 🙂